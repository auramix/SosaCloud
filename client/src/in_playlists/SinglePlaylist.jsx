import styled from 'styled-components';
import React from 'react';
import ArtistAnchorEl from '../styled-comps/ArtistAnchor.jsx';
import { List, Like } from '../styled-comps/TrackInteractions.jsx';
import {Artwork} from '../styled-comps/Artwork.jsx';
import {PlaylistInfo} from './PlaylistInfo.jsx';

const StyledPlaylistSpan = styled.span`
  display: block;
  background: center center/contain no-repeat;
  background-image: url(https://sosacloud-icon-assets.s3-us-west-1.amazonaws.com/playlist_cover.png);
  margin-right: 4px;
  padding: 5px 6px;
  width: 50px;
  height: 50px;
  position: relative;
  float: left;
`;

const StyledDiv = styled.div`
  height: 62px;
  width:306px;
  z-index: 0;
`;

const StyledListItem = styled.li`
  padding: 5px 0px;
`;




export default class SinglePlaylist extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <StyledListItem>
        <StyledDiv >
          <StyledPlaylistSpan>
            <Artwork playlistArtSpan={true} imageUrl={this.props.playlist.playlistImgUrl}/>
          </StyledPlaylistSpan>
          
          <PlaylistInfo playlist={this.props.playlist}/>
          
        </StyledDiv>
      </StyledListItem>
    );
  }
}