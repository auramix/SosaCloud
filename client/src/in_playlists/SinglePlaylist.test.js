import SinglePlaylist from './SinglePlaylist.jsx';

describe('SinglePlaylist', () => {
  it('Has correct number of children', () => {
    const wrapper = shallow( < SinglePlaylist playlist = {
      {
        playlistImgUrl: 'value'
      }
    }/>);

    expect(wrapper.children()).toHaveLength(1);
  });

  // Snapshot test
  it('Has expected snapshot output', () => {
    const wrapper = shallow( < SinglePlaylist playlist = {
        {
          playlistImgUrl: 'value'
        }
      }/>
    );
    expect(wrapper).toMatchSnapshot();
  })

});