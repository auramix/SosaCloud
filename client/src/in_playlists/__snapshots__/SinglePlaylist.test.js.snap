// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SinglePlaylist Has expected snapshot output 1`] = `
<styled.li>
  <styled.div>
    <styled.span>
      <Artwork
        imageUrl="value"
        playlistArtSpan={true}
      />
    </styled.span>
    <PlaylistInfo
      playlist={
        Object {
          "playlistImgUrl": "value",
        }
      }
    />
  </styled.div>
</styled.li>
`;
